src/storage/mod.rs

```diff
@@ -61,7 +61,7 @@ pub fn get_file_list<P: AsRef<Path>>(path: P) -> Result<Vec<PathBuf>, Error> {
 
 pub(crate) enum Storage<'a> {
     Database(DatabaseBackend<'a>),
-    S3(S3Backend<'a>),
+    S3(Box<S3Backend<'a>>),
```

Comment:
What's the reason for making this a `Box`?

Reply:
It was a fix for a clippy lint because of dramatically different variant sizes